src/lib/bcmp/bcmp.cpp

```diff
@@ -99,16 +99,17 @@ static void dfu_copy_and_process_message(struct pbuf *pbuf) {
 */
 int32_t bmcp_process_packet(struct pbuf *pbuf, ip_addr_t *src, ip_addr_t *dst) {
   int32_t rval = 0;
-  // uint8_t src_port;
-  uint8_t dst_port;
+  // uint8_t egress_port;
+  uint8_t ingress_port;
 
   configASSERT(pbuf);
   configASSERT(src);
 
-   // Ingress and Egress ports are mapped to the 5th and 6th byte of the IPv6 src address
-   // as per the bristlemouth protocol spec */
-  // src_port = ((src->addr[1]) & 0xFF);
-  dst_port = ((src->addr[1] >> 8) & 0xFF);
+  // Ingress and Egress ports are mapped to the 5th and 6th byte of the IPv6 src address.
+  // This is in conflict with the v1.0.0, June 2023 version of the Bristlemouth spec.
+  // TODO: align with spec
+  // egress_port = ((src->addr[1]) & 0xFF);
+  ingress_port = ((src->addr[1] >> 8) & 0xFF);
```

Comment:
The spec says bits 108:111 of the source IP address indicates the ingress port. This implementation is using too many bits (8 instead of 4) and in the wrong place (bits 80:87). Similarly, the implementation uses bits 88-95 for the egress port instead of 104:107 in the spec.

Reply:
When we do update this, will sounds like it'll be a breaking release?

Reply:
Yep that's true, so definitely before v1.0 and preferably sooner rather than later. I'd advocate for the release next year of generalized sensor networks, since everyone will have to update at that time anyway.